### PR TITLE
fix: provide pull request to commit parser allow overrides

### LINKED ISSUE #1284: Commit message override is not working
https://github.com/googleapis/release-please#how-can-i-fix-release-notes appears to not be working

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #1284

### CODE REVIEW

File: src/manifest.ts [hunk 1]
```diff
@@ -462,6 +462,7 @@ export class Manifest {
         sha: commit.sha,
         message: commit.message,
         files: commit.files,
+        pullRequest: commit.pullRequest,
```
Comment (bcoe) [inline src/manifest.ts:465]: Is this just the `pullRequest` #, curious why this was an optional parameter?

**Comment by bcoe** [comment]
I'm not sure I'm fully understanding the cause of this bug.

File: src/manifest.ts (hunk 1 above)
Comment (chingor13) [reply src/manifest.ts:465]: This is the full pull request data. It's optional because not all commits on a branch have an associated pull request.

**Comment by chingor13** [comment]
> I'm not sure I'm fully understanding the cause of this bug.

This happened because we weren't passing the pull request data to the strategies when we collected commits. The feature was unit tested assuming that the pull request data was set on the commits.

**Review by bcoe — APPROVED** [review]

**Pull request merged**